### PR TITLE
 Explicitly configure whether or not to use istio ingress

### PR DESCRIPTION
**Pull request by ryant1986**

### CODE REVIEW

File: mmd/compositeconn.go [hunk 1]
```diff
@@ -240,9 +240,25 @@ func (c *CompositeConn) createAndInitDirectConnection(service string) (*ConnImpl
 }
 
 var env = computeEnv()
-var nameserverUrl = env + ".k8s.peak6.net:53"
-var istioIngressUrl = env + ".istioingress.peak6.net"
+var nameserverHost = getEnv("NAMESERVER_HOST", env + ".k8s.peak6.net")
+var istioIngressHost = getEnv("ISTIO_INGRESS_HOST", env + ".istioingress.peak6.net")
 var _, isInK8s = os.LookupEnv("KUBERNETES_SERVICE_HOST")
+var useIstioIngress = getEnvBool("USE_ISTIO_INGRESS", !isInK8s)
```
Comment (yangxizi030) [inline mmd/compositeconn.go:246]: Can we put these variables inside the function use them? so we don't need to evaluate them if those functions don't get called

File: mmd/compositeconn.go [hunk 2]
```diff
@@ -240,9 +240,25 @@ func (c *CompositeConn) createAndInitDirectConnection(service string) (*ConnImpl
...
+func getEnv(key, fallback string) string {
+	if value, ok := os.LookupEnv(key); ok {
+		return value
+	}
+	return fallback
+}
+
+func getEnvBool(key string, fallback bool) bool {
+	if value, ok := os.LookupEnv(key); ok {
+		var result, _ = strconv.ParseBool(value)
```
Comment (yangxizi030) [inline mmd/compositeconn.go:257]: maybe check if parse success here

**Review by yangxizi030 — CHANGES_REQUESTED** [review]

**Review by jlobraco — APPROVED** [review]
lgtm

File: mmd/compositeconn.go (hunk 1 above)
Comment (ryant1986) [reply mmd/compositeconn.go:246]: I think this is preferable to performing the lookups every time they're used

File: mmd/compositeconn.go (hunk 2 above)
Comment (ryant1986) [reply mmd/compositeconn.go:257]: What do you think the right course of action is if it's a parse failure, panic?  I used this approach (consider the value false if it's uparseable), but failing fast for invalid config probably makes more sense..

Comment (jlobraco) [reply mmd/compositeconn.go:257]: I think a panic is acceptable here given the environment variable has been set, but it is not parsable as a bool, we would probably want to know that right away. Nice catch Xizi.

File: mmd/compositeconn.go (hunk 1 above)
Comment (jlobraco) [reply mmd/compositeconn.go:246]: I agree with Ryan here. We could wrap them in a function with a do once block but IMO that would be adding unnecessary indirection. I say we can leave this as it is.

File: mmd/compositeconn.go (hunk 2 above)
Comment (ryant1986) [reply mmd/compositeconn.go]: Done, and moved these methods to mmd.go so that connconfig.go can also use them

**Review by yangxizi030 — APPROVED** [review]
looks good!

**Pull request merged**